### PR TITLE
Idea: add const generic peek_tokens method to parser

### PR DESCRIPTION
**Pull request by jmhain**
This provides a nicer API for peeking multiple tokens ahead.

Also, while this is probably not super impactful in the grand scheme of things since you're generally not peeking more than a few tokens at a time, this technically decreases the complexity of doing so from O(n^2) to O(n).

### CODE REVIEW

**Comment by iffyio** [comment]
LGTM! cc @alamb

**Review by iffyio — APPROVED** [review]

File: src/parser/mod.rs [hunk 1]
```diff
@@ -2895,6 +2893,52 @@ impl<'a> Parser<'a> {
...
+    ///         Token::Word(Word { keyword: Keyword::BY, .. }),
+    ///     ]
+    /// ));
+    /// ```
+    pub fn peek_tokens<const N: usize>(&self) -> [Token; N] {
+        self.peek_tokens_with_location()
+            .map(|with_loc| with_loc.token)
+    }
+
+    pub fn peek_tokens_with_location<const N: usize>(&self) -> [TokenWithLocation; N] {
```
Comment (alamb) [inline src/parser/mod.rs:2923]: Perhaps can we add a docstring to this method too (even if it says "see [`Self::peek_tokens`] for details")?

**Review by alamb — APPROVED** [review]
This is pretty neat -- thank you @jmhain 

I think one more docstring and it will be ready to go

Thanks a lot.

cc @iffyio

File: src/parser/mod.rs (hunk 1 above)
Comment (jmhain) [reply src/parser/mod.rs:2927]: Done

**Pull request merged**